From 90dc44441ecbcf44f3ec80b8bb0ffc87de1a7d92 Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Tue, 4 Jan 2022 19:44:15 -0300
Subject: [PATCH] 85-nm-unmanaged.rules: do not manage docker bridges

Bring back the changes done by
https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/merge_requests/15
by forcing docker bridges to be unmanaged by network-manager.

This is because while they are normal externally managed bridges (and
the cause of the revert of the original change), they can be disabled
and won't be enabled automatically while turning network off and on via
nmcli.

This should be raised with upstream as a bug for finding out a proper
fix.

Upstream-Status: Inappropriate [lmp specific]

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>

---
 data/85-nm-unmanaged.rules | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/data/85-nm-unmanaged.rules b/data/85-nm-unmanaged.rules
index b4f805c..008c7f1 100644
--- a/data/85-nm-unmanaged.rules
+++ b/data/85-nm-unmanaged.rules
@@ -30,6 +30,9 @@ ATTR{address}=="00:1c:42:*", ENV{INTERFACE}=="vnic[0-9]*", ENV{NM_UNMANAGED}="1"
 # LXD situation, let's treat the devices called eth* as regular ethernet.
 ENV{ID_NET_DRIVER}=="veth", ENV{INTERFACE}!="eth[0-9]*", ENV{NM_UNMANAGED}="1"
 
+# Docker bridges. Managed by docker daemon.
+ATTR{address}=="02:42:*", ENV{INTERFACE}=="docker[0-9]*|br-*", ENV{NM_UNMANAGED}="1"
+
 # USB gadget device. Unmanage by default, since whatever created it
 # might want to set it up itself (e.g. activate an ipv4.method=shared
 # connection).
